Manager/src/main/java/io/raspberrywallet/manager/modules/pushbutton/PushButtonModule.java

```diff
@@ -5,17 +5,33 @@
 import com.pi4j.io.gpio.GpioPinDigitalInput;
 import com.pi4j.io.gpio.RaspiPin;
 import com.pi4j.io.gpio.event.GpioPinListenerDigital;
+import io.raspberrywallet.manager.Configuration;
 import io.raspberrywallet.manager.modules.Module;
 
 import static io.raspberrywallet.manager.modules.pushbutton.PushButtonModule.Inputs.PRESSED;
 
-public class PushButtonModule extends Module {
+public class PushButtonModule extends Module<PushButtonConfig> {
 
-    private final GpioController gpio = GpioFactory.getInstance();
-    private final GpioPinDigitalInput pushButton = gpio.provisionDigitalInputPin(RaspiPin.GPIO_23);
+    private final GpioController gpio;
+    private final GpioPinDigitalInput pushButton;
 
-    public PushButtonModule() {
-        super("Press Button");
+    // For test purposes only
+    public PushButtonModule(GpioController gpio) throws InstantiationException, IllegalAccessException {
+        super("Press Button", PushButtonConfig.class);
+        this.gpio = gpio;
+        pushButton = gpio.provisionDigitalInputPin(RaspiPin.GPIO_23);
+    }
+
+    public PushButtonModule() throws InstantiationException, IllegalAccessException {
+        super("Press Button", PushButtonConfig.class);
+        gpio = GpioFactory.getInstance();
+        pushButton = gpio.provisionDigitalInputPin(RaspiPin.GPIO_23);
+    }
+
+    public PushButtonModule(Configuration.ModulesConfiguration modulesConfiguration) throws InstantiationException, IllegalAccessException {
+        super("Press Button", modulesConfiguration, PushButtonConfig.class);
+        gpio = GpioFactory.getInstance();
+        pushButton = gpio.provisionDigitalInputPin(RaspiPin.GPIO_23);
```

Comment:
I can't because there must be test-only constructor PushButtonModule(GpioController gpio) that inject this dependency. I also want to keep the field final, so there can't be default initialization.

Reply:
Ok